libcloudforensics/providers/gcp/internal/common.py

```diff
@@ -187,3 +187,45 @@ def BlockOperation(self, response, zone=None):
       if result['status'] == 'DONE':
         return result
       time.sleep(5)  # Seconds between requests
+
+
+def ExecuteAndPaginate(client, func, kwargs, throttle=False):
+  """Execute and paginate a request to the GCP API.
+
+  Args:
+    client (googleapiclient.Resources): A GCP client object.
+    func (str): A GCP function to query from the client.
+    kwargs (dict): A dictionary of parameters for the function func.
+    throttle (bool): A boolean indicating if requests should be throttled. This
+        is necessary for some APIs (e.g. list logs) as there is an API rate
+        limit. Default is False, i.e. requests are not throttled.
+
+  Returns:
+    list(dict): A list of dictionaries (responses from the request).
+
+  Raises:
+    RuntimeError: If the request to the GCP API could not complete.
+  """
+
+  responses = []
+  next_token = None
+  while True:
+    if throttle:
+      time.sleep(1)
+    if next_token:
+      kwargs['pageToken'] = next_token
+    try:
+      request = getattr(client, func)
+      response = request(**kwargs).execute()
```

Comment:
What is `response` in this case? Can it be a list? if so, you might want to extend instead of append in L227, otherwise you'll end up with a list of lists.

Reply:
My understanding of https://cloud.google.com/compute/docs/reference/rest/v1/ and similar APIs is that it always returns a JSON body as a response, never a list of JSON bodies, so we shouldn't encounter the case you're mentioning.

Reply:
OK cool. The `list` I was anticipating is actually in the `'items`' key of the response; that's where the pagination occurs. Thanks!